LevelComponents/Entity.cpp

```diff
@@ -255,6 +255,100 @@ namespace LevelComponents
+            struct OAMTile *newOAM = new struct OAMTile();
+            newOAM->Xoff = (attr1 & 0xFF) - (attr1 & 0x100); // Offset of OAM tile from entity origin
+            newOAM->Yoff = (attr0 & 0x7F) - (attr0 & 0x80); // they are signed char
+            newOAM->xFlip = (attr1 & (1 << 0xC)) != 0;
+            newOAM->yFlip = (attr1 & (1 << 0xD)) != 0;
+            int SZ = (attr1 >> 0xE) & 3;                         // object size
+            int SH = (attr0 >> 0xE) & 3;                         // object shape
+            newOAM->OAMwidth = OAMDimensions[2 * (SZ * 3 + SH)]; // unit: 8x8 tiles
+            newOAM->OAMheight = OAMDimensions[2 * (SZ * 3 + SH) + 1];
+            int tileID = attr2 & 0x3FF;
+            int palNum = (attr2 >> 0xC) & 0xF;
+
+            // Create the tile8x8 objects
+            for (int y = 0; y < newOAM->OAMheight; ++y)
+            {
+                for (int x = 0; x < newOAM->OAMwidth; ++x)
+                {
+                    struct EntityTile *entityTile = new struct EntityTile();
+                    entityTile->deltaX = x * 8;
+                    entityTile->deltaY = y * 8;
+                    int offsetID = tileID + y * 0x20 + x;
+                    if (offsetID >= 0x200) offsetID = offsetID - 0x200; // the game engine use char Id with offset in the oam data
+                    if (offsetID >= tile8x8data.size()) continue; // this data have problems, just skip it
+                    if (tile8x8data[offsetID] == blankTile) continue; // skip dummy tiles if used
+                    Tile8x8 *newTile = new Tile8x8(tile8x8data[offsetID]);
+                    if (palNum > 7) palNum = palNum - 8; // the game engine use pal Id with offset in the oam data
+                    newTile->SetPaletteIndex(palNum);
+                    entityTile->objTile = newTile;
+                    newOAM->tile8x8.push_back(entityTile);
+                }
+            }
+            tmpOAMTiles.push_back(newOAM);
+        }
+
+        // In oam attributes, x and y params has a range from 0 below 511 and 255
+        // And we want to put the (x, y) = (0, 0) point on the central of the pixmap
+        // So we double the size again
+        int width = 512 * 2; // x
+        int height = 256 * 2; // y
+        QPixmap pm(width, height);
+        pm.fill(Qt::transparent);
+        QPainter p(&pm);
+
+        // OAM tiles must be rendered in reverse order as per the GBA graphical specifications
+        for (auto iter = tmpOAMTiles.rbegin(); iter != tmpOAMTiles.rend(); ++iter)
+        {
+            OAMTile *ot = *iter;
+            // x + 8, y + 16, this works in the room rendering to match the box
+            p.drawImage(ot->Xoff + 512 + 8, ot->Yoff + 256 + 16, ot->Render());
+            delete ot;
+        }
+        tmpOAMTiles.clear();
+
+        // Draw position referrence Box
+        if (!noReferrneceBox)
+        {
+            QPen EntityBoxPen = QPen(QBrush(QColor(0xFF, 0xFF, 0, 0xFF)), 2);
```

Comment:
Let's `#define` the color (and construct the color from an RGBA string instead of components)

Reply:
done.